### PR TITLE
Split aus specific data functions 

### PR DESCRIPTION
**Pull request by aarathybabu97**

### CODE REVIEW

**Comment by njtierney** [comment]
This will resolve #46

File: R/per_capita_household_size.R [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+#' See [get_household_size_distribution()] function for retrieving household size distributions for a given place.
+#' @param household_data data set with information on the household size distribution of specific state or LGA. 
+#' @param household_size bare variable name of the column depicting the household size. Default is 'household_size' from [get_household_size_distribution()].
+#' @param n_people bare variable name of the column depicting the total number of people belonging to the respective household size. Default is 'n_people' from [get_household_size_distribution()]. 
+#' @return Numeric of length 1 - the per capita household size for a given state
+#'     or LGA.
+#' @author Nick Golding
+#' @export
+#' @examples 
+#' sample_data <- get_household_size_distribution(lga = "Fairfield (C)")
```
Comment (njtierney) [inline R/per_capita_household_size.R:12]: Can you print out `sample_data` in the example in the next line, so users can see the columns on the online docs?

File: R/per_capita_household_size.R [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+#' per_capita_household_size(sample_data,household_size,n_people)
+#'
```
Comment (njtierney) [inline R/per_capita_household_size.R:14]: Can you name each argument of this function?

File: R/per_capita_household_size.R [hunk 3]
```diff
@@ -0,0 +1,42 @@
+#' @title Get per capita household size with household size distribution
+#' @description Returns the per capita household size for a location given its household size distribution. 
+#' See [get_household_size_distribution()] function for retrieving household size distributions for a given place.
+#' @param household_data data set with information on the household size distribution of specific state or LGA. 
+#' @param household_size bare variable name of the column depicting the household size. Default is 'household_size' from [get_household_size_distribution()].
```
Comment (njtierney) [inline R/per_capita_household_size.R:5]: I think perhaps it should be `household_size_col`

File: R/per_capita_household_size.R [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+#' @param n_people bare variable name of the column depicting the total number of people belonging to the respective household size. Default is 'n_people' from [get_household_size_distribution()].
```
Comment (njtierney) [inline R/per_capita_household_size.R:6]: perhaps change to `n_people_col`

**Review by njtierney — COMMENTED** [review]
This looks like exactly what we are after, well done! Just a small couple of comments re documentation so far.

**Comment by aarathybabu97** [comment]
`get_per_capita_household_size()` split as `get_household_size_distribution()` and `per_capita_household_size()` (happy to change the name if needed)

`get_household_size_distribution()` : 

```r
library(conmat)
demo_data <- get_household_size_distribution(lga = "Fairfield (C)")
demo_data
#> # A tibble: 8 x 5
#> # Groups:   lga [1]
#>    year state lga           household_size n_people
#>   <dbl> <chr> <chr>                  <dbl>    <dbl>
#> 1  2016 NSW   Fairfield (C)              1     9002
#> 2  2016 NSW   Fairfield (C)              2    26776
#> 3  2016 NSW   Fairfield (C)              3    31599
#> 4  2016 NSW   Fairfield (C)              4    44676
#> 5  2016 NSW   Fairfield (C)              5    33890
#> 6  2016 NSW   Fairfield (C)              6    21216
#> 7  2016 NSW   Fairfield (C)              7     9800
#> 8  2016 NSW   Fairfield (C)              8    10976

demo_state_data <- get_household_size_distribution(state = "NSW")
demo_state_data

#> # A tibble: 1,048 x 5
#> # Groups:   state [1]
#>     year state lga                   household_size n_people
#>    <dbl> <chr> <chr>                          <dbl>    <dbl>
#>  1  2016 NSW   Albury (C)                         1     6020
#>  2  2016 NSW   Albury (C)                         2    13476
#>  3  2016 NSW   Albury (C)                         3     8220
#>  4  2016 NSW   Albury (C)                         4    10164
#>  5  2016 NSW   Albury (C)                         5     5205
#>  6  2016 NSW   Albury (C)                         6     1866
#>  7  2016 NSW   Albury (C)                         7      392
#>  8  2016 NSW   Albury (C)                         8      336
#>  9  2016 NSW   Armidale Regional (A)              1     2983
#> 10  2016 NSW   Armidale Regional (A)              2     7326
#> # ... with 1,038 more rows

```
`per_capita_household_size()` :

```r
per_capita_household_size(
  household_data = demo_data,
  household_size_col = household_size,
  n_people_col = n_people
)

#> [1] 4.199372
per_capita_household_size(
  household_data = demo_state_data,
  household_size_col = household_size,
  n_people_col = n_people
)
#> [1] 3.4407

Created on 2022-05-11 by the reprex package (v2.0.1)


```

**Comment by aarathybabu97** [comment]
@njtierney Would it be better to drop out the arguments `household_size_col` and `n_people_col` in `per_capita_household_size()`  as those are present in the household size distributions output from `get_household_size_distributions() `?  

My reasoning behind adding it was that in case if users had any non-aus household data and therefore did not use the `get_household_size_distribution()`, they should still be able to use the per capita function.

**Comment by njtierney** [comment]
> @njtierney Would it be better to drop out the arguments `household_size_col` and `n_people_col` in `per_capita_household_size()`  as those are present in the household size distributions output from `get_household_size_distributions() `?  

> My reasoning behind adding it was that in case if users had any non-aus household data and therefore did not use the `get_household_size_distribution()`, they should still be able to use the per capita function.

I think they should be kept, as I'm not sure how the `per_capita_household_size` function would work without those arguments? Would they be hard coded then, or?

If people do not have Australian data and didn't use the `get_household_size_distribution()` function then they still need to be able to specify the columns that represent household size and number of people?

**Comment by aarathybabu97** [comment]
> 
> I think they should be kept, as I'm not sure how the `per_capita_household_size` function would work without those arguments? Would they be hardcoded then, or?

Yes. The below hardcoded one would work only when the data used is the output of `get_household_size_distribution()` which  has the column names `household_size` and `n_people`.   
```r
demo_data <- get_household_size_distribution(lga = "Fairfield (C)")
demo_data
#> # A tibble: 8 x 5
#> # Groups:   lga [1]
#>    year state lga           household_size n_people
#>   <dbl> <chr> <chr>                  <dbl>    <dbl>
#> 1  2016 NSW   Fairfield (C)              1     9002
#> 2  2016 NSW   Fairfield (C)              2    26776
#> 3  2016 NSW   Fairfield (C)              3    31599

per_capita_household_size <- function(household_data) {
  household_data %>%
    dplyr::group_by(
      household_size,
      .add = TRUE
    ) %>%
    dplyr::summarise(
      n_people = sum(n_people),
      .groups = "drop"
    ) 
...
rest of the function
```

> If people do not have Australian data and didn't use the `get_household_size_distribution()` function then they still need to be able to specify the columns that represent household size and number of people?

I added the arguments to specify the column names as I thought it would make the function more flexible. If the user has data on non-Australian household size distributions (which wouldn't be an output from `get_household_size_distribution()` as it uses the `abs_household_lga` ), they would still be able to use `per_capita_household_size()`

**Review by njtierney — APPROVED** [review]
Looks good to me!

**Pull request merged**